build_projects/dotnet-host-build/CompileTargets.cs

```diff
@@ -180,19 +176,49 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
                 // Create .rc files on Windows.
                 var resourceDir = GenerateVersionResource(c);
 
+                if (configuration.Equals("Release"))
```

Comment:
I am surprised you had to add this since it would be required independent of Arm64 change.